contracts/poolV2.sol

```diff
@@ -257,7 +263,8 @@ contract Pool is iBEP20 {
       return (outputAmount, fee);
     }
 
-    function burnSynths(address synthIN, address member) public onlyRouter returns(uint outputAmount, uint fee) {
+    function burnSynths(address synthIN, address member) external returns(uint outputAmount, uint fee) {
+    require(iSYNTHFACTORY(_DAO().SYNTHFACTORY()).isSynth(synthIN) == true, "!synth");
```

Comment:
Change from 'public' to 'external' to save gas.
Moved isSynth check to Pool level